src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -25,12 +29,13 @@ public class JwtTokenReader : IJwtTokenReader
         ///     Constructor
         /// </summary>
         /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
-        /// <param name="applicationId">Azure AD Application used as audience to validate against</param>
-        public JwtTokenReader(string applicationId)
+        /// <param name="claimCheck">Azure AD Application used as audience to validate against</param>
+        public JwtTokenReader(IDictionary<string, string> claimCheck)
         {
-            Guard.NotNullOrWhitespace(applicationId, nameof(applicationId));
+            Guard.NotAny(claimCheck, nameof(claimCheck));
```

Comment:
Do you need to guard here on claimChecks for which the value is empty ?

Reply:
The method you mention is a super method of:

```csharp 

public JwtTokenReader (string applicationId): this (new Dictionary <string, string> {{JwtClaimTypes.Audience, applicationId}})

```
therefore the validation is pertinent as long as the method is called with the corresponding signature (parameters). In case it is called through the reference of the base method, it is verified that the applicationid is not null or empty in the respective method